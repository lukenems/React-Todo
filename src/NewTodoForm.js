import React from 'react';

class NewTodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { task: "" }
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.addTodo(this.state);
    this.setState({ task: "" });
  }

  handleChange(evt) {
    this.setState({ [evt.target.name ]: evt.target.value })
  }

  render() {
    return (
    <form onSubmit={this.handleSubmit}>
      <label htmlFor="task">Task:</label>
      <input id="task" name="task" value={this.state.task} onChange={this.handleChange}></input>
      <button>Add a Task</button>
    </form>
    )
  }
}

export default NewTodoForm