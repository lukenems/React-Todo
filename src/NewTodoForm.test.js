import React from "react";
import { shallow, mount} from "enzyme";
import toJson from "enzyme-to-json";
import NewTodoForm from './NewTodoForm';

it("renders without crashing", function() {
  shallow(<NewTodoForm />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<NewTodoForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
})

it("allows for change of new Todo", function() {
  let wrapper = mount(<NewTodoForm />);
  const taskInput = wrapper.find("#task");
  taskInput.instance().value = "chores"
  taskInput.simulate("change");

  expect(wrapper.state().task).toEqual("chores");
});

it("runs a mocked addTodo function on submit", function() {
  const submitFn = jest.fn();
  let wrapper = mount(<NewTodoForm addTodo={submitFn} />);
  const form = wrapper.find("form");

  form.simulate("submit");

  expect(submitFn).toHaveBeenCalled();
});

it("resets state on submit", function() {
  const submitFn = jest.fn();
  let wrapper = mount(<NewTodoForm addTodo={submitFn} />);

  const taskInput = wrapper.find("#task");
  taskInput.instance().value = "chores"
  taskInput.simulate("change");

  expect(wrapper.state().task).toEqual("chores");

  const form = wrapper.find("form");
  form.simulate("submit");

  expect(wrapper.state()).toEqual({
    task: ""
  });
});