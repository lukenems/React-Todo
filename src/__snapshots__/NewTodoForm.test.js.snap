// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="task"
  >
    Task:
  </label>
  <input
    id="task"
    name="task"
    onChange={[Function]}
    value=""
  />
  <button>
    Add a Task
  </button>
</form>
`;
